packages/guides/src/Compiler/NodeTransformers/AnchorNodeTransformer.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Compiler\NodeTransformers;
+
+use phpDocumentor\Guides\Compiler\NodeTransformer;
+use phpDocumentor\Guides\Nodes\AnchorNode;
+use phpDocumentor\Guides\Nodes\DocumentNode;
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\Nodes\TitledNode;
+
+/**
+ * @implements NodeTransformer<Node>
+ *
+ * A custom anchor, also called label, can be applied to sections if directly in front of a headline. It can be applied
+ * directly to a figure as well. In all other cases it is a standalone target rendered separately, which does not have
+ * a title.
+ *
+ * https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-ref
+ */
+class AnchorNodeTransformer implements NodeTransformer
+{
+    /** @var AnchorNode[] */
+    private array $anchorNodes = [];
+
+    public function enterNode(Node $node): Node
+    {
+        if ($node instanceof DocumentNode) {
+            // unset classes when entering the next document
+            $this->anchorNodes = [];
+        }
+
+        if ($node instanceof AnchorNode) {
+            // collect all anchors, multiple anchors can be applied to one section
+            $this->anchorNodes[] = $node;
+        } elseif ($node instanceof TitledNode) {
+            // Titled Nodes handle anchors themselves and provide a title to links
+            foreach ($this->anchorNodes as $anchorNode) {
+                $node->addAnchor((string) $anchorNode->getValue());
```

Comment:
I would like to suggest to add the anchor nodes as child nodes on the title. And remove the anchor from the tree. 

Like you relocate it in the tree... This will simplify the rendering, as we can just render a list of anchors before the parent.

We could introduce a method getAnchors() that will filter the anchors from the children.

The other benefit is that we could remove the getParent method, as this might complicate caching because of circulaire references.

Reply:
Thing is, we cannot always relocate them. If the anchor is directly followed by a new section or a figure or any other titled node that supplies a title for a link target it is directly attached to that one. If any other Node comes next we need to keep it in place. It can be linked to but does not supply a link text.

I think that the section is a more logical location then the title because we are linking to a section, not just the title usually.

We could remove the getParent if that is problematic I guess. I figured we might be able to use it to relocate nodes in a secong preprocessing step if necessary. With the current tools we have I see no other way of manipulating the AST the way that would be needed. If we turned the nodes into a bidirectional graph that would allow us to traverse and manipulate it...